src/dialects/abstract/query-interface.js

```diff
@@ -612,13 +611,21 @@ export class QueryInterface {
    * @returns {Promise}
    */
   async getForeignKeysForTables(tableNames, options) {
+    const generator = this.queryGenerator;
+    tableNames = Array.from([tableNames]).flat(); // force array
```

Comment:
Consider using the `toArray` util we talked about, I see this code very often in your PR

Reply:
Can we look the other way on this? 😏 

Years ago, I used a project that used _underscore_ and then they switched to lodash, which wasn't a huge hit, but was a little work as not everything translated. Later on, they wanted to reduce their library dependency footprint and remove lodash -- you can imagine how that went, since it was integrated into their code.

I hesitate to use lodash where it can be avoided and encourage others to either create wrapper functions or create their custom functions for schemes like the above. But that's my preference and I can make the change if that's your preference and I haven't convinced you.

Reply:
It's not a lodash util (I'm also removing lodash everywhere it's not needed anymore), it's a function we can add to our utilities, see here: https://github.com/sequelize/sequelize/pull/14578#discussion_r885318955

It has better performance (doesn't need to instanciate a few arrays and iterate them to flatten), and is more explicit. Given you've used this `Array.from([tableNames]).flat();` code a few times, I think the util is much needed for clarity

Reply:
The flattening needs to happen regardless as to whether the value is wrapped in an array.  It standardizes the return value and makes it iterable, removing the need for type detection. I only wrapped the value out of paranoia to handle `undefined`s. 

See other comments on `toArray`.  It seems to be equivalent to `Array.from()`, but handles `undefined` and `null`. Unless those are a popular use case, which can clutter the code, I think the custom `toArray` should be removed in favor of the native `Array.from()`.

Perhaps a new function should be created to:
```javascript
function toFlatArray( val ){
  Array.from([ val ]).filter(Boolean).flat()
}
```

Reply:
> The flattening needs to happen regardless as to whether the value is wrapped in an array.

The input cannot be a nested array so the flattening does not need to happen.

> See other comments on `toArray`. It seems to be equivalent to `Array.from()`

It is not, it wraps the value in an array if it is not already an array:

```js
toArray([]) -> []
toArray(null) -> []
toArray(undefined) -> []
toArray(1) -> [1]
toArray([1,2]) -> [1,2]
```